app/controllers/projects/blob_controller.rb

```diff
@@ -146,10 +146,19 @@ def editor_variables
 
     @file_path =
       if action_name.to_s == 'create'
+        if params[:file_upload].present?
+          params[:file_name] = params[:file_upload].original_filename
+        end
         File.join(@path, File.basename(params[:file_name]))
       else
         @path
       end
+     
+    if params[:file_upload].present?
+      params[:content] = params[:file_upload].read
+      params[:encoding] = 'text'
```

Comment:
That would be unfortunate, since there are a lot of binary formats like
doc, ppt, zip, pdf... I think that is the whole point of that feature,
isn't it?

Best regards,
73
On Jul 24, 2015 7:47 PM, "Stan Hu" <notifications@github.com> wrote:

> In app/controllers/projects/blob_controller.rb
> <https://github.com/gitlabhq/gitlabhq/pull/9138#discussion_r35447142>:
>
> >          File.join(@path, File.basename(params[:file_name]))
> >        else
> >          @path
> >        end
> > +
> > +    if params[:file_upload].present?
> > +      params[:content] = params[:file_upload].read
> > +      params[:encoding] = 'text'
>
> First, you should do an md5 check; the files should be different if they
> are binary. Can we just disable the replace/upload functionality for binary
> files?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/gitlabhq/gitlabhq/pull/9138/files#r35447142>.
>